src/sql/connection.py

```diff
@@ -365,14 +369,55 @@ def _get_curr_connection_info(cls):
         }
 
     @classmethod
-    def _transpile_query(cls, query):
-        if not cls.current:
-            return query
+    def _get_curr_sqlglot_dialect(cls):
+        """Get the dialect name in sqlglot package scope
+
+        Returns:
+            str: Available dialect in sqlglot package, see more:
+            https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py
+        """
         connection_info = cls._get_curr_connection_info()
+        if not connection_info:
+            return None
+
+        return DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING.get(
+            connection_info["dialect"], connection_info["dialect"]
+        )
+
+    @classmethod
+    def _is_curr_dialect_support_backtick(cls):
+        """Get if the dialect support backtick (`) syntax as identifier
+
+        Raises:
+            Exception: Indicate there is unknown dialect support
+
+        Returns:
+            bool: If the dialect can use backtick identifier in the SQL clause
+        """
+        cur_dialect = cls._get_curr_sqlglot_dialect()
+        if not cur_dialect:
+            return False
         try:
-            write_dialect = DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING.get(
-                connection_info["dialect"], connection_info["dialect"]
+            return (
+                "`" in sqlglot.Dialect.get_or_raise(cur_dialect).Tokenizer.IDENTIFIERS
             )
+        except Exception as e:
+            raise e
+
+    # TODO: Remove this
+    @classmethod
+    def _transpile_query(cls, query):
+        """Translate the given SQL clause that's compatible to current connected
+        dialect by sqlglot
+
+        Args:
+            query (str): Original SQL clause
+
+        Returns:
+            str: SQL clause that's compatible to current connected dialect
+        """
+        write_dialect = cls._get_curr_sqlglot_dialect()
+        try:
```

Comment:
Good question, we discussed [this before](https://github.com/ploomber/jupysql/pull/164#discussion_r1119452604)

The main reason we don't want just raise the exception is this section relies on external library sqlglot, since the function here if to transpile the original query if possible, we want to safely return the original query when external library has any issues